exonum/src/node/state.rs

```diff
@@ -752,6 +790,25 @@ impl State {
         full_proposes
     }
 
+    /// Checks whether some blocks are waiting for this transaction.
+    /// Returns a list of blocks that don't contain unknown transactions.
+    ///
+    /// Transaction is ignored if the following criteria are fulfilled:
+    ///
+    /// - transaction isn't contained in unknown transaction list of any blocks
+    /// - transaction isn't a part of block
+    pub fn check_incomplete_blocks(&mut self, tx_hash: Hash) -> Vec<(Hash, IncompleteBlock)> {
```

Comment:
Method name is misleading, as it not only checks, but also mutates the state. I'd suggest a name like `remove_unknown_transaction`.